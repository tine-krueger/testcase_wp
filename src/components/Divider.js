import styled from 'styled-components/macro'

export default function Divider({children}) {

    return(
        <DividingHeadline>
            {children}
        </DividingHeadline>
    )
}

const DividingHeadline = styled.h2`
    padding: 1em 0;
    text-transform: uppercase;
    text-align: center;
    background-color: var(--link);
    color: var(--secondaryFont);
`