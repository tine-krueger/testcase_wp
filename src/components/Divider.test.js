import { render } from '@testing-library/react'
import 'jest-styled-components'
import Divider from './Divider'

const children = 'Hello Hamburg'

describe ('Divider', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Divider
                children={children}
            />
        )
        expect(container).toMatchSnapshot()
    })

    it('shows the correct item information', () => {
        const { getByRole } = render(
            <Divider
                children={children}
            />
        )
        expect(getByRole('heading', {name:/Hello Hamburg/i})).toBeInTheDocument()
    })
})