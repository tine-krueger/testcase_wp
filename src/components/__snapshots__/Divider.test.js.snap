// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
.c0 {
  padding: 1em 0;
  text-transform: uppercase;
  text-align: center;
  background-color: var(--link);
  color: var(--secondaryFont);
}

<div>
  <h2
    class="c0"
  >
    Hello Hamburg
  </h2>
</div>
`;
